crates/runtime/src/dataconnector/spiceai.rs

```diff
@@ -57,25 +57,22 @@ use tonic::metadata::MetadataMap;
 
 #[derive(Debug, Snafu)]
 pub enum Error {
-    #[snafu(display("Unable to parse SpiceAI dataset path: {dataset_path}"))]
+    #[snafu(display("Invalid dataset path: {dataset_path}\nEnsure the path is valid"))]
     UnableToParseDatasetPath { dataset_path: String },
 
-    #[snafu(display("Unable to publish data to SpiceAI: {source}"))]
-    UnableToPublishData { source: flight_client::Error },
-
-    #[snafu(display("Missing required parameter: {parameter}"))]
+    #[snafu(display("Missing required parameter: {parameter}. Specify a value.\nFor details, visit: https://docs.spiceai.org/components/data-connectors/spiceai#configuration"))]
     MissingRequiredParameter { parameter: String },
 
-    #[snafu(display(r#"Unable to connect to endpoint "{endpoint}": {source}"#))]
+    #[snafu(display(r#"Failed to connect to SpiceAI endpoint "{endpoint}".\n{source}\nEnsure the endpoint is valid and reachable"#))]
     UnableToVerifyEndpointConnection {
         source: ns_lookup::Error,
         endpoint: String,
     },
 
-    #[snafu(display("Unable to create flight client: {source}"))]
+    #[snafu(display("Failed to create flight client.\n{source}"))]
```

Comment:
why would a Flight client fail to create? what can a user do to resolve it?

if this is an internal error, perhaps direct users to file a bug.